ThePhasesOfCraftship/2_best_practice_first/testingBasics/exercises/1_Red_Green_Refactor/1_2_Palindrome/src/index.spec.ts

```diff
@@ -1,4 +1,35 @@
+import { isPalindrome } from "./index";
 
-describe('palindrome checker', () => {
+describe("palindrome checker", () => {
+  test("should confirm that 'mom' is palindrome", () => {
+    expect(isPalindrome("mom")).toBeTruthy();
+  });
 
-})
\ No newline at end of file
+  test.each(["Mom", "MoM"])(
+    "should confirm %s is palindrome when is single uppercase word",
+    (val) => {
+      expect(isPalindrome(val)).toBeTruthy();
+    }
+  );
+
+  test.each(["Momx", "rocket", "JohnDeep"])(
+    "should not confirm %s as palindrome",
+    (val) => {
+      expect(isPalindrome(val)).toBeFalsy();
+    }
+  );
+
+  test.each(["Was It A Rat I Saw", "Never Odd or Even", "1Never Odd or Even1"])(
+    "should confirm phrase (%s) as palindrome ",
+    (val) => {
+      expect(isPalindrome(val)).toBeTruthy();
+    }
+  );
+
+  test.each(["That's not a palindrome", "Never Odd or Even1"])(
+    "should not confirm phrase (%s) as palindrome ",
+    (val) => {
+      expect(isPalindrome(val)).toBeFalsy();
+    }
+  );
+});
```

Comment:
Your tests are great, but I think we can simplify it. Maybe you can have just two test suites, one for palindromes and one for non-palindromes. This way the business logic is tested and we hide the integration details and nonsense like saying a capitalized word, non-capitalized word and that stuff. I hope you understand what I'm saying. Nevertheless great job!

Reply:
Mario, this palindrome problem is a bit easy and I agree with you.
>capitalised word, non-capitalised word and stuff like that
could be treated as implementation logic, but on the other hand if we want to keep small increment without jumping to the end of the solution I have feeling that it's a good approach to treat above details as business one. Happy to hear your thoughts 🙌🏿

Reply:
Hi Mateusz. I would keep the implementation details in tests while working towards the end goal, which is to test the feature and I don't see anything bad in testing it during the development process however when refactoring the tests at the end I would remove them as they clutter the testing suite and don't provide a lot of value